Week1 [Step3] namukim

Groot-94 opened the pull request:
@uuu1101  안녕하세요 테테님! 
열심히 피드백 해주셔서 코딩과 순서도를 만들면서 조금 더 집중하고 좋은 결과를 만들어 가는거 같아서 굉장히 기분이 좋습니다..!
이번엔 README 파일에 순서도를 넣어봤습니다..! 신기하네요 ㅎㅎㅎ 오늘도 좋은하루 보내시고 피드백 주시면 감사하겠습니다!

- 어려웠던 점:

하면서 어려웠던 부분은 아무래도 누드빼빼로를 만드는 과정이 쉽게 떠오르지 않아서 여러번 코드를 고쳤습니다..
결국에 생각한 방법이 몸통을 그리는 함수에 nudeBebero 라는 bool 타입의 매개변수를 추가해서 누드빼빼로에 대해서만 좀 특별한 그림을 그리도록 작성해봤습니다.

- 궁금한 점:

길이(Int)에 따른 막대(빼빼로 손잡이)를 그리는 함수 -> 1번 함수
몸통(String), 토핑(String)을 매개변수로 받아 토핑과 몸통을 그리는 함수 -> 2번 함수
길이(Int)에 따라 몸통을 그리는 함수 -> 3번 함수
각종 매개변수를 받아 위의 세 메서드를 호출하는 함수 -> 4번 함수
이렇게 이름을 붙이겠습니다!

 2번 함수가 직접 print문을 사용해서 출력한다고 했을때,  3번 함수를 이용해서 몸통을 그리는 방법이 생각보다 떠오르지 않았습니다..
그래서 저는 2번 함수를 String을 리턴하는 형식을 사용했습니다.

__저기 작성해주신 "그리는 함수" 라는 부분이 정말 print문을 사용하는 함수를 말씀하시는지 궁금합니다 :)__

Groot-94 committed:
Step3 빼빼로 만들기
first commit

uuu1101 commented on CodeStarterCamp_Week1/main.swift:
```diff
@@ -1,29 +1,47 @@
-        }
-        print("")
+//길이(Int)에 따른 막대(빼빼로 손잡이)를 그리는 함수
+func drawStic(Height: Int) {
```
이름이 명확하네요. 
`stic`을 `stick`으로 변경하시는건 어떨까요?!
발음으로 의미를 유추 할 수 있지만, 영어표기법에 맞는 단어를 사용하는 것도 중요하답니다 ! 
저도 단어가 기억이 안나거나 함수명 만들 때 올바른지 찾아보곤 한답니다 :) 
영어는 참 어렵네요 😭

Groot-94 replied:
정확한 영어표기법을 놓치고 있었네요ㅜ 감사합니다!

uuu1101 commented on CodeStarterCamp_Week1/main.swift:
```diff
@@ -1,29 +1,47 @@
-        print("","|","|",  separator:"   ")
-    }
+//몸통(String), 토핑(String)을 매개변수로 받아 토핑과 몸통을 리턴하는 함수
+func selectBody(body: String, tofing: String) -> String {
```
함수명을 보고 뭔가 Body를 선택할 것 같았는데 Body와 tofing을 조합하여 하나의 형태로 **만들어**주는 것 같네요.
더 좋은 이름이 있을까요?! 🤔
`tofing`을 `topping`으로 변경해보는건 어떨까요?! 😊

Groot-94 replied:
makeTopBody 라는 이름으로 변경해봤습니다..

uuu1101 commented on CodeStarterCamp_Week1/main.swift:
```diff
@@ -1,29 +1,47 @@
+func makeBebero (bodyHeight: Int, bodyType: String, tofingType: String, sticHeight: Int) {
+    let nude : Bool
+    print("<정보>\n길이: \(bodyHeight)\n몸통: \(bodyType)\n토핑: \(tofingType)\n막대길이: \(sticHeight)\n")
+    if bodyType == "|0|" {
+        nude = true
+    } else {
+        nude = false
+    }
```
누드 빼뺴로인지 체크하는 부분을 넣어주셨네요 👍

uuu1101 reviewed (commented):
@starting1231 안녕하세요! 
순서도를 잘 표현해주셨네요 !
**현재 로직에서 `Body`에 아무것도 넣지 않고 입력하면 어떻게 될까요?! 👀**

- 토핑이 있는 빼빼로의 경우엔 잘 출력이 되지만,
토핑이 없는 애는 토핑에 `" "` 빈 값을 넣어주어야 하는데요. 
이 값을 주지 않고 원하는 결과값을 나오게는 할 수 없을까요?! 🤔

과제에 나와있는 조건을 그대로 빈 값`""`을 넣었을 때 출력 되는 값은 아래와 같습니다.
```Swift
<정보>
길이: 7
몸통: ***
토핑: 
막대길이: 6

***
***
***
***
***
***
***
 | | 
 | | 
 | | 
 | | 
 | | 
 | | 
```
- 과제 구현 사항에 3가지 함수를 호출하는 함수를 만들라고 되어 있는데, 
현재는 2가지 함수를 호출 하는 것 같아요. 어떻게 바꿔보면 좋을까요?!

Groot-94 committed:
Feat : Topping이 "" 일 때, Body 양옆에 공백 추가

Refactor : 함수 및 매개변수 명 수정, 함수호출 변경
 - stic -> stick
 - tofing -> topping
 - selectBody -> makeTopBody
 - makeTopBody 함수를 BeberoBody 변수에서 호출

uuu1101 commented on CodeStarterCamp_Week1/main.swift:
```diff
@@ -8,15 +8,20 @@
-    return ("\(tofing)\(body)\(tofing)")
+func makeTopBody(body: String, topping: String) -> String {
+    if topping == "" {
+        return (" \(body) ")
```
문자열 보간법에 대해서 찾아보신것 같네요! 👍
`(" \(body) ")` 뒤에 있는 공백은 제거하셔도 될 것 같습니다 :) 
`("\(body)") 또한 양 옆에 있는 괄호는 지워주어도 의미가 같아요. 괄호를 생략해서 표현 한답니다

uuu1101 commented on CodeStarterCamp_Week1/main.swift:
```diff
@@ -8,15 +8,20 @@
+        return (" \(body) ")
+    }
+    else {
+        return ("\(topping)\(body)\(topping)")
```
아래 46번째 줄은 `} else {` 로 작성하셨는데, 여기는 줄바꿈을 하셨네요 👀
줄바꿈도 하나의 의도를 담을 수 있어요. 따라서 일관적이게 작성하시는게 좋아보여요 ! 
통상적으로 `} else {`라고 표현합니다. 😊

uuu1101 commented on CodeStarterCamp_Week1/main.swift:
```diff
@@ -32,16 +37,18 @@ func drawBody(Height: Int, body: String, nudeBebero: Bool) {
 //각종 매개변수를 받아 위의 세 메서드를 호출하는 함수
-func makeBebero (bodyHeight: Int, bodyType: String, tofingType: String, sticHeight: Int) {
+func makeBebero (bodyHeight: Int, bodyType: String, toppingType: String, stickHeight: Int) {
     let nude : Bool
```
`let nude : Bool` 과 ` let nude : Bool` 둘 중 어느것이 더 보기 좋을까요? 🤔
통상적으로 `let nude: Bool` 이라고 작성하고 있어요. 
관련하여 Swift API Guideline 이나 Swift 공식문서를 보시면 좋을 것 같습니다.

uuu1101 commented on CodeStarterCamp_Week1/main.swift:
```diff
@@ -1,29 +1,54 @@
+            print(" | | ")
+        }
+        else{
+            print(body)
```
여기 `else` 구문도 위 코멘트와 동일한부분입니다. 😊

uuu1101 reviewed (commented):
@starting1231 나무님! 주말은 잘 보내셨나요?!
nude라는 Bool값으로 누드빼빼로를 체크 하는 부분이 인상적이었어요 👏

몇가지 피드백을 남겨 두었어요. 
해당 부분에 대해서 고민해보시고 commit 남겨주세요 :) 
고생하셨습니다 !

Groot-94 committed:
Refactor : 불필요한 문자 제거, 매개변수명 수정, 줄바꿈 수정
- 13, 20, 30, 39, 40, 48번 줄 공백제거
- Height 매개변수 -> height 수정
- 21, 31번 else 문 줄바꿈 수정
